Adds Typing Indicator

Kapu1178 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Based on the Baystation12 typing indicator:

When a human is in the process of typing via the "say" or "me" windows, they will gain a little speech bubble adjacent to their sprite indicating they are in the process of typing.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Encourages more verbal interaction. Players are less likely to walk off or do something else if they know someone they can see is typing.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: Humans now display a chat bubble when typing, as an effort to move us closer to Baystation
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Kapu1178 committed:
Ports Typing Indicator

francinum commented:
I am advancing my agenda, one step at a time 😏 

But really, this thing is kinda only vaguely reliable and is probably going to result in some pretty shitty play for a few days.

PowerfulBacon commented:
Who do me verbs have a typing indicator?

Lagomorphica commented:
finally I know when to attack after asking somebody a question 😈

Kapu1178 commented:
> 
> 
> Who do me verbs have a typing indicator?

me's are just as important as normal speech, and we should be encouraging our players to use it more.

PowerfulBacon commented:
I didn't even know there was a hotkey for *me, who doesn't just type them in the command line to avoid the dialogs 😅

ivanmixo commented:
> finally I know when to attack after asking somebody a question 😈

this is why we can't have nice things

Kapu1178 committed:
fixes dme

BrumalLake commented:
andd here's the guy who said that making the icon a var for all mobs was stinky a couple of days ago

Kapu1178 commented:
> 
> 
> andd here's the guy who said that making the icon a var for all mobs was stinky a couple of days ago

not what i said at all. i said putting it in human/Initialize() was stinky, but it looks like I had no choice.

Kapu1178 commented:
known bug: Sticky overlays happen when you attempt to open a say dialog while you already have a say dialog open

Kapu1178 committed:
Fixes sticky overlay

francinum commented on code/modules/mob/typing_indicator.dm:
```diff
@@ -1,8 +1,9 @@
 /mob/proc/create_typing_indicator()
```
...If this only works on human why is this not a stub function and an override defined on `mob/living/carbon/human`

Kapu1178 replied:
Because I’m filter worded. Will do when I get home

Kapu1178 committed:
Removes Kapu code

Kapu1178 committed:
Some changes

Moved everything to typing_indicators.dm, and moved the human mutable appearance to a global datum

Kapu1178 committed:
No longer appears while anything but conscious

qwertyquerty commented:
I don't like this. :key:

Kapu1178 committed:
Merge branch 'master' into typing_indicator

Kapu1178 commented:
pulled master to unfuck DME file
